rs/index/src/segment/pending_segment.rs

```diff
@@ -44,13 +50,28 @@ impl<Q: Quantizer + Clone + Send + Sync> PendingSegment<Q> {
             .map(|segment| segment.name())
             .collect();
 
+        let temp_invalidated_ids_directory = format!("{}/temp_invalidated_ids_storage", parent_directory);
```

Comment:
I don't think we should use `parent_directory`. I think it should be `data_directory`. Let's write a test to confirm?